File: Minimum Integer.cpp
class Solution {
  public:
    int minimumInteger(int N, vector<int> &A) {
        long long sum = 0;
        int mini = INT_MAX;
        for(auto it : A)
         sum += it;
         
         for(int i = 0; i < N; i++){
             if((long long)A[i]*N >= sum){
                 mini = min(mini,A[i]);
             }
         }
         return mini;
    }
};

File: Minimum Integer.java
class Solution {
    public static int minimumInteger(int N, int[] A) {
         
         long sum=0; 
         int min=Integer.MAX_VALUE;
         
         for(int i=0;i<A.length;i++)
         {
             sum=sum+A[i];
         }
         
         for(int i=0;i<A.length;i++)
         {
             if(sum<=(long)A.length*A[i])
             {
                 if(A[i]<min)
                   min=A[i];
             } 
         }
         
         return min;
         
    }
}
